packages/block-editor/src/components/block-tools/zoom-out-mode-inserter-button.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { plus } from '@wordpress/icons';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+function ZoomOutModeInserterButton( {
+	previousClientId,
+	nextClientId,
+	index,
+} ) {
+	const {
+		sectionRootClientId,
+		setInserterIsOpened,
+		hasSelection,
+		selectedBlockClientId,
+		hoveredBlockClientId,
+	} = useSelect( ( select ) => {
+		const {
+			getSettings,
+			getSelectionStart,
+			getSelectedBlockClientId,
+			getHoveredBlockClientId,
+		} = select( blockEditorStore );
+		const { sectionRootClientId: root } = unlock( getSettings() );
+
+		return {
+			hasSelection: !! getSelectionStart().clientId,
+			sectionRootClientId: root,
+			setInserterIsOpened:
+				getSettings().__experimentalSetIsInserterOpened,
+			selectedBlockClientId: getSelectedBlockClientId(),
+			hoveredBlockClientId: getHoveredBlockClientId(),
+		};
+	}, [] );
```

Comment:
Seems like some of this could be passed into the component from `ZoomOutModeInserters` rather than doing the work again here.